mypy/messages.py

```diff
@@ -875,22 +875,29 @@ def typeddict_instantiated_with_unexpected_items(self,
                                                      expected_item_names: List[str],
                                                      actual_item_names: List[str],
                                                      context: Context) -> None:
-        self.fail('Expected items {} but found {}.'.format(
-            expected_item_names, actual_item_names), context)
+        if not expected_item_names:
+            expected = '(no keys)'
+        else:
+            expected = format_key_list(expected_item_names)
+        found = format_key_list(actual_item_names, short=True)
+        if actual_item_names and set(actual_item_names) < set(expected_item_names):
+            found = 'only {}'.format(found)
+        self.fail('Expected {} but found {}'.format(expected, found), context)
 
     def typeddict_item_name_must_be_string_literal(self,
                                                    typ: TypedDictType,
                                                    context: Context,
                                                    ) -> None:
-        self.fail('Cannot prove expression is a valid item name; expected one of {}'.format(
-            format_item_name_list(typ.items.keys())), context)
+        self.fail(
+            'Cannot prove expression is a valid TypedDict key; expected one of {}'.format(
```

Comment:
"Cannot prove" is still pretty unfriendly. I think this can just be "TypedDict key must be a string literal", since this message appears when it's not a string literal (visit_typeddict_index_expr in checkexpr.py).